openwisp_controller/subnet_division/rule_types/base.py

```diff
@@ -54,9 +54,7 @@ def provision_receiver(cls, instance, **kwargs):
 
     @classmethod
     def destroyer_receiver(cls, instance, **kwargs):
-        transaction.on_commit(
-            lambda: cls.destroy_provisioned_subnets_ips(instance, **kwargs)
-        )
+        cls.destroy_provisioned_subnets_ips(instance, **kwargs)
```

Comment:
Since deleting subnets was being triggered `on_comit`, the `on_delete=model.CASCADE` set on config field of `SubnetDivisionRuleIndex` was deleting the entries of the subnets to be deleted from the table, before our handler could get executed. This was resulting in empty queryset(for fetching subnets to be deleted). 

Removing that handler from `transaction.on_comit` solved the issue. I have added automated tests to ensure that.